pennylane/hf/dipole.py

```diff
@@ -0,0 +1,308 @@
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.42525091, 0.62391373, 0.1688554],
+    >>>                   [3.42525091, 0.62391373, 0.1688554]], requires_grad=True)
+    >>> mol = qml.hf.Molecule(symbols, geometry, alpha=alpha)
+    >>> args = [alpha]
+    >>> coeffs, ops = fermionic_dipole(mol)(*args)[2]
+    >>> ops
+    [[], [0, 0], [0, 2], [1, 1], [1, 3], [2, 0], [2, 2], [3, 1], [3, 3]]
+    """
+
+    def _fermionic_dipole(*args):
+        r"""Compute the fermionic dipole moment observable.
+
+        Args:
+            args (array[array[float]]): initial values of the differentiable parameters
+
+        Returns:
+            tuple(array[float], list[list[int]]): the dipole moment coefficients and operators
+        """
+        f = []
+        constants, integrals = dipole_integrals(mol, core, active)(*args)
+        constants = anp.negative(constants)
+
+        for i in range(3):
+            f.append(fermionic_one(constants[i], integrals[i], cutoff=cutoff))
+
+        return f
+
+    return _fermionic_dipole
+
+
+def dipole_moment(mol, cutoff=1.0e-12, core=None, active=None):
+    r"""Return a function that computes the qubit dipole moment observable.
+
+    Args:
+        mol (Molecule): the molecule object
+        cutoff (float): cutoff value for discarding the negligible electronic integrals
+        core (list[int]): indices of the core orbitals
+        active (list[int]): indices of the active orbitals
+
+    Returns:
+        function: function that computes the qubit dipole moment observable
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.42525091, 0.62391373, 0.1688554],
+    >>>                   [3.42525091, 0.62391373, 0.1688554]], requires_grad=True)
+    >>> mol = qml.hf.Molecule(symbols, geometry, alpha=alpha)
+    >>> args = [alpha]
+    >>> dipole_moment(mol)(*args)[2].terms[1]
+    [PauliZ(wires=[0]),
+     PauliY(wires=[0]) @ PauliZ(wires=[1]) @ PauliY(wires=[2]),
+     PauliX(wires=[0]) @ PauliZ(wires=[1]) @ PauliX(wires=[2]),
+     PauliZ(wires=[1]),
+     PauliY(wires=[1]) @ PauliZ(wires=[2]) @ PauliY(wires=[3]),
+     PauliX(wires=[1]) @ PauliZ(wires=[2]) @ PauliX(wires=[3]),
+     PauliZ(wires=[2]),
+     PauliZ(wires=[3])]
+    """
+
+    def _dipole(*args):
+        r"""Compute the qubit dipole moment observable.
+
+        Args:
+            args (array[array[float]]): initial values of the differentiable parameters
+
+        Returns:
+            (list[Hamiltonian]): x, y and z components of the dipole moment observable
+        """
+        d = []
+        d_ferm = fermionic_dipole(mol, cutoff, core, active)(*args)
+        for i in d_ferm:
+            d.append(qubit_operator(i, cutoff=cutoff) * (-1))
+
+        return d
+
+    return _dipole
+
+
+def fermionic_one(core_constant, integral, cutoff=1.0e-12):
```

Comment:
This function can be used for computing other observables, e.g., the Hamiltonian, and will be moved to a more appropriate place in the future _reorganizing_ PRs.

Reply:
Yes, there will be a separate PR at the end to reorganize functions.